pkg/k8sclient/k8s_apiextensions_client.go

```diff
@@ -0,0 +1,94 @@
+package k8sclient
```

Comment:
How is this significantly different from [the other client factory implementation](https://github.com/openshift/assisted-service/blob/3541ee5294a89fe9a10a03f1b561790a30a10cc2/internal/spoke_k8s_client/spoke_k8s_client.go) we already have? Why create a new one for this?

Reply:
This is using specifically the apiextensions [clientset](k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset). Can try to make the SpokeK8sClient more generic and get a specific clientset. Just not sure it really worth the effort, as it serves different use-cases:/